toxav/toxav.c

```diff
@@ -225,12 +249,48 @@ Tox *toxav_get_tox(const ToxAV *av)
 {
     return av->tox;
 }
+
+uint32_t toxav_audio_iteration_interval(const ToxAV *av)
+{
+    return av->calls ? av->audio_stats.interval : IDLE_ITERATION_INTERVAL_MS;
+}
+
+uint32_t toxav_video_iteration_interval(const ToxAV *av)
+{
+    return av->calls ? av->video_stats.interval : IDLE_ITERATION_INTERVAL_MS;
+}
+
 uint32_t toxav_iteration_interval(const ToxAV *av)
 {
-    /* If no call is active interval is 200 */
-    return av->calls ? av->interval : 200;
+    return min_u32(toxav_audio_iteration_interval(av),
+                   toxav_video_iteration_interval(av));
 }
-void toxav_iterate(ToxAV *av)
+
+/**
+ * @brief calc_interval Calculates the needed iteration interval based on previous decode times
+ * @param av ToxAV struct to work on
+ * @param stats Statistics to update
+ * @param frame_time the duration of the current frame in ms
+ * @param start_time the timestamp when decoding of this frame started
+ */
+static void calc_interval(ToxAV *av, DecodeTimeStats *stats, int32_t frame_time, uint64_t start_time)
+{
+    stats->interval = frame_time < stats->average ? 0 : (frame_time - stats->average);
+    stats->total += current_time_monotonic(av->m->mono_time) - start_time;
+
+    if (++stats->count == 3) {
+        stats->average = stats->total / 3 + 5 /* NOTE Magic Offset for precision */;
```

Comment:
Cimple doesn't support comments inside expressions like this. Put the ; before the comment.